.github/workflows/ci-workflow.yml

```diff
@@ -0,0 +1,73 @@
+name: ciworkflow
+
+on:
+  push:
+    branches:
+      - '*'
```

Comment:
Does this mean the workflow is triggered on push to any branches? If so, 

- Why do we need `!master` below?
- I do not think we want to trigger workflow on non-master branches unless there is a PR opened for that branch.

Reply:
Removed.
(1) While testing I have added !master and forgot to remove.
(2) Changed the filter from `push` to `pull_request` so that it will build only for PR's.

Reply:
But this does not trigger workflow for push to `master` branch. I think it should be like this.
```
on:
  push:
    branches:
      - master
  pull_request:
    branches:
      - master
```